intel_pytorch_extension_py/__init__.py

```diff
@@ -31,11 +31,80 @@ def get_auto_optimization():
 def enable_auto_mix_precision(mixed_dtype = torch.bfloat16):
     if mixed_dtype == torch.bfloat16:
         core.enable_mix_bf16_fp32()
+        core.disable_mix_int8_fp32()
+    elif mixed_dtype == torch.int8 or mixed_dtype == torch.uint8:
+        core.enable_mix_int8_fp32()
+        core.disable_mix_bf16_fp32()
     else:
+        core.disable_mix_int8_fp32()
         core.disable_mix_bf16_fp32()
 
 def get_auto_mix_precision():
     if core.get_mix_bf16_fp32():
         return torch.bfloat16
+    elif core.get_mix_int8_fp32():
+        return torch.int8
     else:
         return None
+
+def calibration_reset():
+    if core.get_int8_calibration():
+        core.calibration_reset()
+    else:
+        raise ValueError("please first run enable_calibration before calibration reset")
+
+class _DecoratorContextManager:
+    """Allow a context manager to be used as a decorator, copy form pytorch FW"""
+
+    def __call__(self, func):
+        if inspect.isgeneratorfunction(func):
+            return self._wrap_generator(func)
+
+        @functools.wraps(func)
+        def decorate_context(*args, **kwargs):
+            with self:
+                return func(*args, **kwargs)
+        return decorate_context
+
+    def _wrap_generator(self, func):
+        """Wrap each generator invocation with the context manager"""
+        @functools.wraps(func)
+        def generator_context(*args, **kwargs):
+            gen = func(*args, **kwargs)
+            while True:
+                try:
+                    with self:
+                        x = next(gen)
+                    yield x
+                except StopIteration:
+                    break
+        return generator_context
+
+class int8_calibration(_DecoratorContextManager):
+    def __init__(self, file_name):
+        self.configure_file = file_name
+
+    def __enter__(self):
+        if not core.get_mix_int8_fp32():
+            raise ValueError("please first run enable_auto_mix_precision(torch.int8) before int8 calibration")
+        core.enable_int8_calibration()
+
+    def __exit__(self, *args):
+        core.disable_int8_calibration()
+        core.add_indicators()
+        core.save_indicators_file(self.configure_file)
+        return False
+
+class int8_validate(_DecoratorContextManager):
+    def __init__(self, file_name):
+        self.configure_file = file_name
+
+    def __enter__(self):
+        if not core.get_mix_int8_fp32():
+            raise ValueError("please first run enable_auto_mix_precision(torch.int8) before int8 validation")
+        core.disable_int8_calibration()
+        core.load_indicators_file(self.configure_file)
```

Comment:
How about loading the quantization configuration from a memory object instead of a file? We can define separate API for serializing and deserializing the memory object to/from a file. This can separate the concerns of loading/saving the configuration and serialization of it.